src/main/proto/com/kodypay/grpc/pay/v1/pay.proto

```diff
@@ -31,6 +31,24 @@ message PayRequest {
   string amount = 2; // amount in BigDecimal/2.dp (0.00)
   string terminal_id = 3; // send the payment to this terminal serial number
   optional bool show_tips = 4; // Show tips on the terminal
+  optional PaymentMethod payment_method = 5; // Choose specific payment method on the terminal. If not set the terminal will request on the screen to select the payment method
+}
+
+
+
+message PaymentMethod{
+  PaymentMethodType payment_method_type = 1; // Payment method type
+  // Verification mode is needed for AliPay and WeChat.
+  oneof verification_mode {
```

Comment:
Can a oneof be optional?
I just realised that if `payment_method = CARD` this property is not necessary.

Reply:
According to the documentation:

"Oneof fields are like optional fields except all the fields in a oneof share memory, and at most one field can be set at the same time." so I believe we are covered.

Also kp-core will do validation for every use case

Reply:
In that case we are not covered at all.
You need to find a way where with CARDS option doesn't require any other setting

Reply:
The wording for oneof "at most one" means that they are both optional to begin with and oneof makes sure that if one is set the other isnt. Apart from that they behave like regular optionals.